### DISCUSSION TITLE
Telegram isn't automatically updating with Homebrew.

### DISCUSSION THREAD

**Discussion by HalideMedia** [Homebrew/discussions#5030] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.2
ORIGIN: https://github.com/Homebrew/brew
HEAD: 705d2564ddfb9c915ee9425dd9ed408bfac61005
Last commit: 5 days ago
Core tap JSON: 06 Jan 23:13 UTC
Core cask tap JSON: 06 Jan 23:13 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: octa-core 64-bit arm_blizzard_avalanche
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.2.1-arm64
CLT: 15.1.0.0.1.1700200546
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  openssl@1.1
```

### Description of issue

For some reason, homebrew isn't automatically updating Telegram ... even though I installed it via "brew install telegram" and run "brew update && brew upgrade" every time I log in to my MacBook Air.

Even after the upgrade, when I run "brew info telegram" I see:

```shell
==> telegram: 10.5,257870 (auto_updates)
https://macos.telegram.org/
/opt/homebrew/Caskroom/telegram/10.4,257123 (122B)
From: https://github.com/Homebrew/homebrew-cask/blob/HEAD/Casks/t/telegram.rb
==> Name
Telegram for macOS
==> Description
Messaging app with a focus on speed and security
==> Artifacts
Telegram.app (App)
==> Analytics
install: 3,119 (30 days), 10,461 (90 days), 29,768 (365 days)
```

Which I believe means that "brew update" found Telegram 10.5 but "brew upgrade" didn't install it, and my MacBook still has Telegram 10.4 installed. This is only confirmed when I start Telegram and see the "Upgrade Telegram" button in the lower-left of the app, as in the screenshot below.

<img width="320" alt="Screenshot 2024-01-06 at 6 39 35 PM" src="https://github.com/Homebrew/discussions/assets/115757904/f46b3ca3-0515-486f-beec-e1bfa11c5df3">

Who knows what else isn't up to date? Can anyone tell me how to fix this?

**Comment by ZhongRuoyu** [comment]
https://docs.brew.sh/FAQ#why-arent-some-apps-included-during-brew-upgrade

If you want to upgrade it with `brew`, use `brew upgrade --greedy telegram`.
